src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -2596,6 +2596,20 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         if( arr_descr->is_array(llvm_tmp) ) {
             llvm_tmp = builder->CreateLoad(arr_descr->get_pointer_to_data(llvm_tmp));
         }
+
+        // TODO: refactor this into a function, it is being used a few times
+        llvm::Type *target_type = llvm_tmp->getType();
+        // Create alloca to get a pointer, but do it
+        // at the beginning of the function to avoid
+        // using alloca inside a loop, which would
+        // run out of stack
+        llvm::BasicBlock &entry_block = builder->GetInsertBlock()->getParent()->getEntryBlock();
+        llvm::IRBuilder<> builder0(context);
+        builder0.SetInsertPoint(&entry_block, entry_block.getFirstInsertionPt());
+        llvm::AllocaInst *target = builder0.CreateAlloca(
+            target_type, nullptr, "call_arg_value_ptr");
```

Comment:
Yes, it was needed because we needed to pass a pointer, but we only had a value.

Reply:
This happens for example if you do `f(3)`, where the argument expects a pointer, but you pass in a number (value).

Then since you need `alloca`, you have to do it at the beginning of the function, because if `f(3)` is inside some loop, then the `alloca` will keep allocating on stack and it runs out of stack and segfaults.

Reply:
Well for passing a number to a function accepting `CPtr` type I will have to write a code something like,

```python
p: CPtr
p_c_pointer(pointer(3), p)
f(p)
```

Now, `pointer(3)` doesn't make sense to me. Or am I missing something here?

Reply:
For pointer you don't need it, I agree. You need it when passing integer constants (or any result of an expression, which gets loaded by default by the `visit_expr` visitors), into a function call, which expects a pointer (that is how we pass arguments by default).

Reply:
i.e., the following should work,

```python
@ccall
def f(x: CPtr)
   pass

f(3)
```

Reply:
No, it should not work, that should be a compiler error. 

This should work, and does:
```
def f(x: i32)
   ...

f(3)
```
And the above code is needed (at another place in LLVM).

For CPtr, it seemed to me it was needed, but your PR also makes things work, and it is not needed.

Reply:
I see then we don't need to worry about creating a different variable inside `GetPointer`. I think `pointer(3)` should be an error as well.